src/Spice86/ViewModels/MemoryViewModel.cs

```diff
@@ -115,16 +115,14 @@ private void SetSearchDataType(MemorySearchDataType searchDataType) {
     public string? StartAddress {
         get => _startAddress;
         set {
-            if (ValidateAddressRange(_state,value, EndAddress,
-                nameof(StartAddress))) {
-                IsMemoryRangeValid = true;
-                if (SetProperty(ref _startAddress, value)) {
-                    TryUpdateHeaderAndMemoryDocument();
-                }
-            } else {
-                IsMemoryRangeValid = false;
-                DataMemoryDocument = null;
+            ValidateAddressRange(_state, value, EndAddress,
+                nameof(StartAddress));
+            IsMemoryRangeValid = true;
+            if (SetProperty(ref _startAddress, value)) {
+                TryUpdateHeaderAndMemoryDocument();
             }
+            IsMemoryRangeValid = false;
+            DataMemoryDocument = null;
```

Comment:
Are you sure we want to do this every time StartAddress is set?

Reply:
No, I'm re-thinking this bit of code. This should be a CanExecute.

Reply:
Done.